Admin/CrossTenantMailboxMigrationValidation.ps1

```diff
@@ -0,0 +1,1052 @@
+.#>
+
+param (
+    [Parameter(Mandatory = $True, ParameterSetName = "ObjectsValidation", HelpMessage = "Validate source Mailbox and Target MailUser objects. If used alone you will be prompted to introduce the identities you want to validate")]
+    [Parameter(Mandatory = $False, ParameterSetName = "OfflineMode", HelpMessage = "Validate source Mailbox and Target MailUser objects. If used alone you will be prompted to introduce the identities you want to validate")]
+    [System.Management.Automation.SwitchParameter]$CheckObjects,
+    [Parameter(Mandatory = $False, ParameterSetName = "ObjectsValidation", HelpMessage = "Path pointing to the CSV containing the identities to validate. CheckObjects parameter needs also to be specified")]
+    [Parameter(Mandatory = $True, ParameterSetName = "CollectMode", HelpMessage = "Path pointing to the CSV containing the identities to validate. CheckObjects parameter needs also to be specified")]
+    [System.String]$CSV,
+    [Parameter(Mandatory = $True, HelpMessage = "Path pointing to the log file")]
+    [System.String]$LogPath,
+    [Parameter(Mandatory = $True, ParameterSetName = "OrgsValidation", HelpMessage = "Validate the organizations settings like organization relationships, migraiton endpoint and AADApplication")]
+    [Parameter(Mandatory = $False, ParameterSetName = "OfflineMode", HelpMessage = "Validate the organizations settings like organization relationships, migraiton endpoint and AADApplication")]
+    [System.Management.Automation.SwitchParameter]$CheckOrgs,
+    [Parameter(Mandatory = $True, ParameterSetName = "SDP", HelpMessage = "Collect relevant data for troubleshooting purposes and send it to Microsoft Support if needed")]
+    [System.Management.Automation.SwitchParameter]$SDP,
+    [Parameter(Mandatory = $False, ParameterSetName = "CollectMode", HelpMessage = "Collect source only mode, to generate the necessary files and provide them to the target tenant admin. You need to specify the CSV parameter as well")]
+    [System.Management.Automation.SwitchParameter]$CollectSourceOnly,
+    [Parameter(Mandatory = $True, ParameterSetName = "SDP", HelpMessage = "Path that will be used to store the collected data")]
+    [Parameter(Mandatory = $True, ParameterSetName = "CollectMode", HelpMessage = "Path that will be used to store the collected data")]
+    [Parameter(Mandatory = $True, ParameterSetName = "OfflineMode", HelpMessage = "Path that will be used to store the collected data, you should specify the path and the zip file name")]
+    [System.String]$PathForCollectedData,
+    [Parameter(Mandatory = $false, ParameterSetName = "OfflineMode", HelpMessage = "Do not connect to source EXO tenant, but specify a zip file gathered when running the script with the 'CollectSourceOnly' parameter.")]
+    [System.Management.Automation.SwitchParameter]$SourceIsOffline
+)
+
```

Comment:
```suggestion

. $PSScriptRoot\..\Shared\OutputOverrides\Write-Host.ps1
```

Let's add the Write-Host output override to this script. This will ensure that when the script calls Write-Host with -ForegroundColor Red, we change the color to the ErrorForegroundColor set by the user. This can be helpful for people who have changed their PowerShell colors to be more readable.

See https://github.com/microsoft/CSS-Exchange/tree/main/Shared/OutputOverrides for more details on these overrides.